src/scylla/db.clj

```diff
@@ -439,6 +445,18 @@
     (when (cu/exists? disabled)
       (c/exec :mv disabled scylla-bin))))
 
+(defn wait-for-previous [test node]
+  (let [nodes (:nodes test)
+        fst (first nodes)]
+    (when (not= node fst)
+        (loop [[previous & tail] nodes]
+            (if (= node (first tail))
+                (do
+                  (info "Waiting for" previous)
+                  (sc/close! (sc/await-open test previous))
```

Comment:
qWhat does this do exactly? How do you wait tha tthe previous node is started? The check got to be more sophisticated - I recommend grepping the log for 'serving', or opening a CQL connection and executing a test query, or both. You can't be too cautious here, we have had lots of flakiness earlier when we declared half-started nodes ready.

Reply:
> opening a CQL connection and executing a test query, or both

I think that's exactly what `sc/await-open` does:
```clojure
(defn await-open
  "Blocks until a connection is available, then returns that connection."
  [test node]
  (dt/with-retry [tries 20]
    (let [c (open test node)]
      (alia/execute (:session c)
                    (hayt/->raw (hayt/select :system.peers)))
      c)
    (catch NoHostAvailableException e
      (when (zero? tries)
        (throw+ {:type :jepsen.db/setup-failed
                 :msg  :await-open-timeout
                 :node node}))
      (info node "not yet available, retrying")
      (Thread/sleep await-open-interval)
      (retry (dec tries)))))

```

Reply:
As Kamil said, ```sc/await-open``` waits by sending test queries.